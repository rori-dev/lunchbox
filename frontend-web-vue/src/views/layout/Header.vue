<template>
  <header
    class="sm:absolute sm:top-0 sm:right-0
           sm:px-8 pt-4 pb-2 sm:pb-4"
  >
    <nav
      class="flex items-center justify-center
             sm:h-16"
      aria-label="Haupt"
    >
      <ul class="list-none">
        <NavLink to="/" exact>
          <OffersIcon />
        </NavLink>

        <NavLink to="/settings">
          <SettingsIcon />
        </NavLink>

        <NavLink to="/about">
          <InfoIcon />
        </NavLink>
      </ul>
    </nav>
  </header>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import OffersIcon from '@/assets/icons/offers.svg'
import SettingsIcon from '@/assets/icons/settings.svg'
import InfoIcon from '@/assets/icons/info.svg'
import NavLink from '@/views/layout/header/NavLink.vue'

@Component({
  components: {
    NavLink,
    OffersIcon,
    SettingsIcon,
    InfoIcon,
  },
})
export default class Header extends Vue {}
</script>
