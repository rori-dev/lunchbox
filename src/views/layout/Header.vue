<template>
  <div
    class="flex flex-wrap justify-between items-center
           px-4 py-4"
  >
    <DaySelector class="w-full sm:max-w-sm" />

    <nav
      class="flex justify-center
             w-full sm:w-auto
             order-first sm:order-last
             px-4 pb-4 sm:pb-0"
      aria-label="Main"
    >
      <router-link to="/" exact v-slot="{ navigate, isActive }">
        <button
          class="flex justify-center items-center
                 w-12 h-12
                 mx-2"
          :class="isActive ? 'text-gray-500' : 'text-gray-900'"
          @click="navigate"
          title="Mittagsangebote öffnen"
          aria-label="Mittagsangebote öffnen"
        >
          <OffersIcon class="w-8 h-8" />
        </button>
      </router-link>
      <router-link to="/settings" v-slot="{ navigate, isActive }">
        <button
          class="flex justify-center items-center
                 w-12 h-12
                 mx-2"
          :class="isActive ? 'text-gray-500' : 'text-gray-900'"
          @click="navigate"
          title="Einstellungen öffnen"
          aria-label="Einstellungen öffnen"
        >
          <SettingsIcon class="w-8 h-8" />
        </button>
      </router-link>
      <router-link to="/about" v-slot="{ navigate, isActive }">
        <button
          class="flex justify-center items-center
                 w-12 h-12
                 mx-2"
          :class="isActive ? 'text-gray-500' : 'text-gray-900'"
          @click="navigate"
          title="Info öffnen"
          aria-label="Info öffnen"
        >
          <InfoIcon class="w-8 h-8" />
        </button>
      </router-link>
    </nav>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import LocationSelector from './LocationSelector.vue'
import DaySelector from '@/views/offers/DaySelector.vue'
import OffersIcon from '@/assets/icons/offers.svg'
import SettingsIcon from '@/assets/icons/settings.svg'
import InfoIcon from '@/assets/icons/info.svg'

@Component({
  components: {
    DaySelector,
    LocationSelector,
    OffersIcon,
    SettingsIcon,
    InfoIcon,
  },
})
export default class Header extends Vue {}
</script>
