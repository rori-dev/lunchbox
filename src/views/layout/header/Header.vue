<template>
  <b-navbar toggleable="sm" type="light">

    <b-navbar-brand to="/">
      <img alt="Vue logo" width="32" src="@/assets/lunchbox_logo.png">
      Lunchbox
    </b-navbar-brand>

    <b-navbar-toggle target="nav_collapse"></b-navbar-toggle>

    <b-collapse is-nav id="nav_collapse">
        <b-navbar-nav>
          <LocationSelector/>
        </b-navbar-nav>

        <!-- Right aligned nav items -->
        <b-navbar-nav class="ml-auto">
            <b-nav-item to="/" exact>Mittagsangebote</b-nav-item>
            <b-nav-item to="/settings" class="d-block d-sm-none">Einstellungen</b-nav-item>
            <b-nav-item to="/about">Info</b-nav-item>
        </b-navbar-nav>
    </b-collapse>

  </b-navbar>
</template>



<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import LocationSelector from './LocationSelector.vue'

@Component({
  components: {
    LocationSelector,
  },
})
export default class Header extends Vue {
}
</script>
