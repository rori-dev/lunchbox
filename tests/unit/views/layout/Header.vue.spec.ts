import Header from '@/views/layout/Header.vue'
import { mountUnit } from '@tests/unit/test-util'

describe('Header', () => {
  test('renders snapshot', () => {
    const wrapper = mountUnit(Header)
    expect(wrapper.element).toMatchSnapshot()
  })
/*
  test('has nav items in collapse', () => {
    const wrapper = mountUnit(Header)

    const collapse = wrapper.find(BCollapse)
    expect(collapse.exists()).toBeTrue()
    const navitems = collapse.findAll(BNavItem)
    expect(navitems.at(0).text()).toBe('Mittagsangebote')
    expect(navitems.at(1).text()).toBe('Einstellungen')
    expect(navitems.at(2).text()).toBe('Info')
  })*/
})
