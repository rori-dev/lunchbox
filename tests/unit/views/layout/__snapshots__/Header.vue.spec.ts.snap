// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders snapshot 1`] = `
<div
  class="flex flex-wrap justify-between items-center
         px-4 py-4"
>
  <dayselector-stub
    class="w-full sm:max-w-sm"
  />
   
  <nav
    aria-label="Main"
    class="flex justify-center
           w-full sm:w-auto
           order-first sm:order-last
           px-4 pb-4 sm:pb-0"
  >
    <router-link-stub
      event="click"
      exact="true"
      tag="a"
      to="/"
    />
     
    <router-link-stub
      event="click"
      tag="a"
      to="/settings"
    />
     
    <router-link-stub
      event="click"
      tag="a"
      to="/about"
    />
  </nav>
</div>
`;
